test/specs/lib/factories-test.tsx

```diff
@@ -264,7 +265,72 @@ describe('factories', () => {
           Component: 'p',
           overrideProps,
           render(Component, props) {
-            expect(props.styles).toMatchObject({
+            expect(callable(props.styles)()).toMatchObject({
+              position: 'keep',
+              color: 'black',
+              ':hover': {
+                position: 'keep',
+                color: 'blue',
+              },
+            })
+          },
+        })
+      })
+
+      test('deep merges styles prop as function onto defaultProps styles', () => {
+        expect.assertions(1)
+
+        const defaultProps = {
+          styles: () => ({
+            color: 'override me',
+            ':hover': { color: 'blue' },
+          }),
+        }
+        const props = {
+          styles: { color: 'black' },
+        }
+
+        getShorthand({
+          value: props,
+          Component: 'p',
+          defaultProps,
+          render(Component, props) {
+            expect(callable(props.styles)()).toMatchObject({
+              color: 'black',
+              ':hover': { color: 'blue' },
+            })
+          },
+        })
+      })
+
+      test('deep merges overrideProps styles as function onto styles prop', () => {
+        expect.assertions(1)
+
+        const overrideProps = {
```

Comment:
it might be a bit easier to reason about expected results if these two `props` objects will be swapped

Reply:
I would rather leave this tests the same as the previous ones for consistency, as the only difference is changing the objects to functions.